src/main/java/com/wynntils/modules/utilities/overlays/inventories/GuildTerritoryManageOverlay.java

```diff
@@ -0,0 +1,117 @@
+            int nameWidth = McIf.mc().fontRenderer.getStringWidth(McIf.getUnformattedText(e.getGui().getUpperInv().getDisplayName()));
+            searchField = new GuiTextFieldWynn(201, McIf.mc().fontRenderer, nameWidth + 13, 108, 157 - nameWidth, 10);
+            searchField.setText("Search...");
+        }
+
+        Keyboard.enableRepeatEvents(true);
+    }
+
+    @SubscribeEvent
+    public void onClose(GuiOverlapEvent.ChestOverlap.GuiClosed e) {
+        // reset everything
+        searchField = null;
+        inTerritoryManageMenu = false;
+        Keyboard.enableRepeatEvents(false);
+    }
+
+    @SubscribeEvent(priority = EventPriority.LOWEST)
+    public void onDrawBackground(GuiOverlapEvent.ChestOverlap.DrawGuiContainerBackgroundLayer e) {
+        if (!inTerritoryManageMenu) return;
+
+        // search highlight
+        for (Slot s : e.getGui().inventorySlots.inventorySlots) {
+            if (s.getStack().isEmpty() || !s.getStack().hasDisplayName() || (s.getStack().getItem() != Items.MAP && s.getStack().getItem() != Items.PAPER)) continue;
+            if (!s.getStack().getDisplayName().startsWith(TextFormatting.WHITE + TextFormatting.BOLD.toString())) continue;
+            String displayName = StringUtils.stripControlCodes(s.getStack().getDisplayName()).toLowerCase(Locale.ROOT);
+            if (searchField.getText().isEmpty() || !displayName.contains(searchField.getText().toLowerCase(Locale.ROOT))) continue;
+
+            SpecialRendering.renderGodRays(e.getGui().getGuiLeft() + s.xPos + 5,
+                    e.getGui().getGuiTop() + s.yPos + 6, 0, 5f, 35, UtilitiesConfig.INSTANCE.guildTerritoryMenuSearchHighlightColor);
+        }
+    }
+
+    @SubscribeEvent(priority = EventPriority.LOWEST)
+    public void onDrawForeground(GuiOverlapEvent.ChestOverlap.DrawGuiContainerForegroundLayer e) {
+        if (!inTerritoryManageMenu) return;
+
+        ScreenRenderer.beginGL(0, 0);
+        GlStateManager.translate(0, 0, 300F);
+        if (searchField != null) searchField.drawTextBox();
+        GlStateManager.translate(0, 0, -300F);
+        ScreenRenderer.endGL();
+    }
+
+    @SubscribeEvent
+    public void onMouseClicked(GuiOverlapEvent.ChestOverlap.MouseClicked e) {
+        if (!inTerritoryManageMenu) return;
+
+        int offsetMouseX = e.getMouseX() - e.getGui().getGuiLeft();
+        int offsetMouseY = e.getMouseY() - e.getGui().getGuiTop();
+
+        // handle mouse input on search box
+        if (searchField != null) {
+            searchField.mouseClicked(offsetMouseX, offsetMouseY, e.getMouseButton());
+            if (e.getMouseButton() == 0) { // left click
+                if (searchField.isFocused()) {
+                    searchField.setCursorPositionEnd();
+                    searchField.setSelectionPos(0);
+                } else {
+                    searchField.setSelectionPos(searchField.getCursorPosition());
+                }
+            }
+        }
```

Comment:
This looks an "if hell" that can be easily cleaned up, whenever you are checking for a linear inverted condition use returns to make everything keep in the same line! Also perhaps most of this logic could be done inside the mouseClicked on the `searchField`! Be careful with this code suggestion tho, I made it on the web interface so spacing might be wrong 🤔 

```suggestion
        if (searchField == null) {
            return;
        }
        
        // Process the search field click interaction
        searchField.mouseClicked(offsetMouseX, offsetMouseY, e.getMouseButton());
        
        // Only listen for left clicks
        if (e.getMouseButton() != 0) {
            return;
        }
        
        if (searchField.isFocused()) {
            searchField.setCursorPositionEnd();
            searchField.setSelectionPos(0);
            return;
        }
        
        searchField.setSelectionPos(searchField.getCursorPosition());
```